auth-foundation/src/main/java/com/okta/authfoundation/client/internal/NetworkUtils.kt

```diff
@@ -117,6 +122,45 @@ internal suspend fun <T> OidcConfiguration.internalPerformRequest(
     }
 }
 
+private suspend fun OidcConfiguration.executeRequest(
+    request: Request
+): Response {
+    currentCoroutineContext().ensureActive()
+    val rateLimitErrorCode = 429
+
+    var response = okHttpClient.newCall(request).await()
+
+    if (response.code == rateLimitErrorCode) {
+        val event = RateLimitExceededEvent(request, response)
+        eventCoordinator.sendEvent(event)
+
+        for (i in 1..event.maxRetries) {
+            val timeToResetSeconds: Long
+            val requestId: String?
+            response.use {
+                val responseTimeSeconds = response.headers.getDate("Date")?.time?.milliseconds?.inWholeSeconds
+                val retryTimeSeconds = response.header("X-Rate-Limit-Reset")?.toLongOrNull()
+                requestId = response.header("X-Okta-Request-Id")
```

Comment:
What happens if multiple of any of these headers exist? `response.headers().values("foo")` returns a list (HTTP allows multiple response headers with the same key. One of your recent test additions sets multiple response headers, which made me think of this.

Reply:
Good question. Going to look further into that. I didn't realize earlier that I was setting multiple response headers with the same key. I thought it was replacing the key, but didn't check the response in detail.

Reply:
In this case, the value of the header appearing last in the response is taken. Also, the setHeader in OkHttp replaces the currently set value if the header already exists in the response, so I'm not testing that at the moment. Picking the last value in the response is something OkHttp is doing. So, is that something I should add a test for?

Reply:
Sounds good. Thanks for looking into this! I'm fine without an explicit test for this 👍